mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestAccessKey.java

```diff
@@ -0,0 +1,144 @@
+import org.mycore.datamodel.metadata.MCRObjectID;
+import org.mycore.mcr.acl.accesskey.MCRAccessKeyManager;
+import org.mycore.mcr.acl.accesskey.MCRAccessKeyTransformer;
+import org.mycore.mcr.acl.accesskey.model.MCRAccessKey;
+import org.mycore.mcr.acl.accesskey.restapi.v2.annotation.MCRRequireAccessKeyAuthorization;
+import org.mycore.mcr.acl.accesskey.restapi.v2.model.MCRAccessKeyInformation;
+import org.mycore.restapi.annotations.MCRRequireTransaction;
+import org.mycore.restapi.v2.MCRErrorResponse;
+
+@Path("/objects/{" + PARAM_MCRID + "}/accesskeys")
+public class MCRRestAccessKey {
+
+    private static final String VALUE = "value";
+
+    private WebApplicationException getUnknownObjectException() {
+        return MCRErrorResponse.fromStatus(Response.Status.NOT_FOUND.getStatusCode())
+            .withMessage("Object doesn't exists!")
+            .withErrorCode("objectNotFound")
+            .toException();
+    }
+    
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @MCRRequireAccessKeyAuthorization
+    public Response getAccessKeys(@PathParam(PARAM_MCRID) final MCRObjectID objectId,
+        @DefaultValue("0") @QueryParam("offset") long offset,
+        @DefaultValue("128") @QueryParam("limit") long limit) {
+        if (!MCRMetadataManager.exists(objectId)) {
+            throw getUnknownObjectException();
+        }
+        final long fromIndex = offset;
+        List<MCRAccessKey> accessKeys = MCRAccessKeyManager.getAccessKeys(objectId);
+        final int totalAccessKeyCount = accessKeys.size();
+        if (offset < 0 || limit <= 0 || fromIndex >= Integer.MAX_VALUE || totalAccessKeyCount == 0 
+            || (int) fromIndex >= totalAccessKeyCount) {
+            accessKeys = new ArrayList<MCRAccessKey>();
+        } else {
+            long toIndex = offset + limit;
+            if (toIndex > Integer.MAX_VALUE) {
+                toIndex = Integer.MAX_VALUE;
+            }
+            if ((int) toIndex >= totalAccessKeyCount) {
+                toIndex = (long) totalAccessKeyCount;
+            }
+            accessKeys = accessKeys.subList((int) fromIndex, (int) toIndex);
+        }
+        return Response.ok(new MCRAccessKeyInformation(accessKeys, totalAccessKeyCount)).build();
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @MCRRequireAccessKeyAuthorization
+    @MCRRequireTransaction
+    public Response addAccessKey(@PathParam(PARAM_MCRID) final MCRObjectID objectId, final String accessKeyJson) {
+        final MCRAccessKey accessKey = MCRAccessKeyTransformer.accessKeyFromJson(accessKeyJson);
+        if (!MCRMetadataManager.exists(objectId)) {
+            throw getUnknownObjectException();
+        }
+        accessKey.setObjectId(objectId);
+        accessKey.setCreator(null);
+        accessKey.setCreation(null);
+        accessKey.setLastChanger(null);
+        accessKey.setLastChange(null);
```

Comment:
Why is that?

Reply:
To protect that the values can be manipulated via the REST API. The manager doesn't do that.